MaxText/layers/quantizations.py

```diff
@@ -83,7 +83,15 @@ def _rhs_axis_metadata_wrapper(
     no_sharding_axis: Sequence[int],
     mesh_axes: Tuple[str, ...],
     is_tiled: bool,
+    replicate_scale: bool = False,
 ):
+  if replicate_scale:
```

Comment:
Please add more comments on why this flag is needed and when it should be used.

Reply:
As discussed - suggest breaking this into separate PRs, sharding changes, quantization and config changes. Quantization change should also be reviewed by aqt team, as there may be a better way to replicate scale supported in code.